pkg/syncer/broker/syncer.go

```diff
@@ -136,7 +136,7 @@ type Syncer struct {
 }
 
 // NewSyncer creates a Syncer that performs bi-directional syncing of resources between a local source and a central broker.
-func NewSyncer(config SyncerConfig) (*Syncer, error) {
+func NewSyncer(config *SyncerConfig) (*Syncer, error) {
```

Comment:
This is passed by value b/c it modifies `config` although it doesn't really need to. In general, it's obviously more efficient to pass by reference but that does open up the possibility of the struct being unexpectedly mutated which might be an unwanted side effect for the  caller. In this case I think it's fine to keep this pass-by-value since it's only called once at startup so any performance hit is negligible. If we want to change it, we just have to make sure this function doesn't mutate `config`.

Reply:
Right, so since this is the only change, it probably doesn’t make sense to do this in Admiral ;-).